core/src/main/java/com/scalar/db/storage/jdbc/RdbEngineOracle.java

```diff
@@ -451,4 +502,35 @@ public void throwIfAlterColumnTypeNotSupported(DataType from, DataType to) {
               from.toString(), to.toString()));
     }
   }
+
+  @Override
+  public void bindBlobColumnToPreparedStatement(
+      PreparedStatement preparedStatement, int index, byte[] bytes) throws SQLException {
+    // When writing to the BLOB data type with a BLOB size greater than 32766 using a MERGE INTO
+    // statement, an internal error ORA-03137 on the server side occurs so we needed to use a
+    // workaround. This has been confirmed to be a limitation by AWS support.
+    // Below is a detailed explanation of the workaround.
+    //
+    // Depending on the byte array size, the JDBC driver automatically choose one the following mode
+    // to transfer the BLOB data to the server:
+    // - DIRECT: the most efficient mode. It's used when the byte array length is less than 32767.
+    // - STREAM: this mode is less efficient. It's used when the byte array length is greater than
+    // 32766.
+    // - LOB BINDING: this mode is the least efficient. It's used when an input stream without
+    // specifying the length is specified.
+    //
+    // When the driver selects the STREAM mode, the error
+    // ORA-03137 occurs. So, we work around the issue by making sure to use the driver in a way so
+    // that it should never selects the STREAM mode.
+    // For more details about the modes, see the following documentation:
+    // https://docs.oracle.com/en/database/oracle/oracle-database/23/jjdbc/LOBs-and-BFiles.html#GUID-8FD40D53-8D64-4187-9F6F-FF78242188AD
+    if (bytes.length <= 32766) {
+      // the DIRECT mode is used to send BLOB data of small size
+      preparedStatement.setBytes(index, bytes);
+    } else {
+      // the LOB BINDING mode is used to send BLOB data of large size
+      InputStream inputStream = new ByteArrayInputStream(bytes);
+      preparedStatement.setBinaryStream(index, inputStream);
+    }
```

Comment:
As the comments explain, there is a limitation with a large BLOB file when executing a ScalarDB Put operation that gets translated to a `MERGE INTO` SQL statement. This limitation requires special handling to put BLOB data. 
I added several integration tests in `core/src/integration-test/java/com/scalar/db/storage/jdbc/JdbcDatabaseColumnValueIntegrationTest.java` to test the values that could be problematic for the implemented workaround.